cli/cmd/check.go

```diff
@@ -51,18 +53,25 @@ non-zero exit code.`,
 	cmd.Args = cobra.NoArgs
 	cmd.PersistentFlags().StringVar(&options.versionOverride, "expected-version", options.versionOverride, "Overrides the version used when checking if Linkerd is running the latest version (mostly for testing)")
 	cmd.PersistentFlags().BoolVar(&options.preInstallOnly, "pre", options.preInstallOnly, "Only run pre-installation checks, to determine if the control plane can be installed")
+	cmd.PersistentFlags().BoolVar(&options.dataPlaneOnly, "proxy", options.dataPlaneOnly, "Only run data-plane checks, to determine if the data plane is healthy")
 	cmd.PersistentFlags().BoolVar(&options.wait, "wait", false, "Retry and wait for some checks to succeed if they don't pass the first time")
 
 	return cmd
 }
 
 func configureAndRunChecks(options *checkOptions) {
 	checks := []healthcheck.Checks{healthcheck.KubernetesAPIChecks}
-	if options.preInstallOnly {
-		checks = append(checks, healthcheck.LinkerdPreInstallChecks)
+
+	if options.dataPlaneOnly {
```

Comment:
Related to @dadjeibaah's comment, my thinking about the data plane check is that it's going to require a working control plane, especially for things like #1517. So my preference is to run both `KubernetesAPIChecks` and `LinkerdAPIChecks` before running `LinkerdDataPlaneChecks`. Maybe something like this would be clearer:

```
	var checks []healthcheck.Checks

	if options.preInstallOnly {
		checks = []healthcheck.Checks{
			healthcheck.KubernetesAPIChecks,
			healthcheck.LinkerdPreInstallChecks,
			healthcheck.LinkerdVersionChecks,
		}
	} else if options.dataPlaneOnly {
		checks = []healthcheck.Checks{
			healthcheck.KubernetesAPIChecks,
			healthcheck.LinkerdAPIChecks,
			healthcheck.LinkerdDataPlaneChecks,
		}
	} else {
		checks = []healthcheck.Checks{
			healthcheck.KubernetesAPIChecks,
			healthcheck.LinkerdAPIChecks,
			healthcheck.LinkerdVersionChecks,
		}
	}
```

Reply:
yeah, true! good point. I think since the k8s api check is always required, we should have slightly less repetition and do 
```
	checks := []healthcheck.Checks{healthcheck.KubernetesAPIChecks}

	if options.preInstallOnly {
		checks = append(checks, healthcheck.LinkerdPreInstallChecks)
		checks = append(checks, healthcheck.LinkerdVersionChecks)
	} else if options.dataPlaneOnly {
		checks = append(checks, healthcheck.LinkerdPreInstallChecks)
		checks = append(checks, healthcheck.LinkerdDataPlaneChecks)
	} else {
		checks = append(checks, healthcheck.LinkerdAPIChecks)
		checks = append(checks, healthcheck.LinkerdVersionChecks)
	}
```